crates/bevy_ecs/src/query/filter.rs

```diff
@@ -93,6 +93,11 @@ pub unsafe trait QueryFilter: WorldQuery {
     ///
     /// Note that this is called after already restricting the matched [`Table`]s and [`Archetype`]s to the
     /// ones that are compatible with the Filter's access.
+    /// If the implementation of [`WorldQuery::matches_component_set`] is incorrect,
+    /// this method may never be called, and your query filter may fail to check entities that should be included.
+    ///
+    /// Implementors of this method will generally either have a trivial `true` body (for archetypal filters),
```

Comment:
One thing that wasn't clear to me at first that may be worth saying explicitly: 

If `IS_ARCHETYPAL` is `true`, then `filter_fetch` must *always* return `true`.